tests/unit/archive_validate_test.py

```diff
@@ -0,0 +1,266 @@
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 0",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=10,
+                    hash="hash0",
+                ),
+                "resource1": Resource(
+                    name="resource1",
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 1",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=20,
+                    hash="hash1_created",
+                ),
+            },
+            [
+                validate.FileDiff(
+                    name="resource1",
+                    diff_type="CREATE",
+                    size_diff=20,
+                )
+            ],
+        ),
+        (
+            {
+                "resource0": Resource(
+                    name="resource0",
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 0",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=10,
+                    hash="hash0",
+                ),
+                "resource1": Resource(
+                    name="resource1",
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 1",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=20,
+                    hash="hash1",
+                ),
+            },
+            {
+                "resource0": Resource(
+                    name="resource0",
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 0",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=10,
+                    hash="hash0",
+                ),
+                "resource1": Resource(
+                    name="resource1",
+                    path="https://www.fake.link",
+                    remote_url="https://www.fake.link",
+                    title="Resource 1",
+                    parts={},
+                    mediatype="zip",
+                    format="format",
+                    bytes=20,
+                    hash="hash1",
+                ),
+            },
+            [],
+        ),
+    ],
+)
+def test_process_resource_diffs(baseline_resources, new_resources, diffs):
+    """Test resource diffs."""
+    test_diffs = validate._process_resource_diffs(baseline_resources, new_resources)
```

Comment:
Typically I'd avoid testing the private function and test the public interface instead - here, that'd be `RunSummary.create_summary`. 

It's 6PM on a Friday so no code sample, but I'll crank on one on Monday morning so you have something to actually compare things to.

Reply:
```diff
diff --git a/tests/unit/archive_validate_test.py b/tests/unit/archive_validate_test.py
index 46eb31a..7b332b9 100644
--- a/tests/unit/archive_validate_test.py
+++ b/tests/unit/archive_validate_test.py
@@ -59,8 +59,8 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
     "baseline_resources,new_resources,diffs",
     [
         (
-            {
-                "resource0": Resource(
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -71,7 +71,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -82,9 +82,9 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=20,
                     hash="hash1",
                 ),
-            },
-            {
-                "resource0": Resource(
+            ],
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -95,7 +95,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -106,7 +106,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=40,
                     hash="hash1_changed",
                 ),
-            },
+            ],
             [
                 validate.FileDiff(
                     name="resource1",
@@ -116,8 +116,8 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
             ],
         ),
         (
-            {
-                "resource0": Resource(
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -128,7 +128,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -139,9 +139,9 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=20,
                     hash="hash1",
                 ),
-            },
-            {
-                "resource0": Resource(
+            ],
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -152,7 +152,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-            },
+            ],
             [
                 validate.FileDiff(
                     name="resource1",
@@ -162,8 +162,8 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
             ],
         ),
         (
-            {
-                "resource0": Resource(
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -174,9 +174,9 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-            },
-            {
-                "resource0": Resource(
+            ],
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -187,7 +187,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -198,7 +198,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=20,
                     hash="hash1_created",
                 ),
-            },
+            ],
             [
                 validate.FileDiff(
                     name="resource1",
@@ -208,8 +208,8 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
             ],
         ),
         (
-            {
-                "resource0": Resource(
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -220,7 +220,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -231,9 +231,9 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=20,
                     hash="hash1",
                 ),
-            },
-            {
-                "resource0": Resource(
+            ],
+            [
+                Resource(
                     name="resource0",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -244,7 +244,7 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=10,
                     hash="hash0",
                 ),
-                "resource1": Resource(
+                Resource(
                     name="resource1",
                     path="https://www.fake.link",
                     remote_url="https://www.fake.link",
@@ -255,12 +255,24 @@ def test_process_partition_diffs(baseline_partitions, new_partitions, diffs):
                     bytes=20,
                     hash="hash1",
                 ),
-            },
+            ],
             [],
         ),
     ],
 )
-def test_process_resource_diffs(baseline_resources, new_resources, diffs):
+def test_process_resource_diffs(baseline_resources, new_resources, diffs, mocker):
     """Test resource diffs."""
-    test_diffs = validate._process_resource_diffs(baseline_resources, new_resources)
+    baseline = mocker.MagicMock(
+        resources=baseline_resources, version="0.0.1", created="2023-01-01"
+    )
+    new = mocker.MagicMock(
+        resources=new_resources, version="0.0.2", created="2023-01-02"
+    )
+    summary = validate.RunSummary.create_summary(
+        name="test package",
+        baseline_datapackage=baseline,
+        new_datapackage=new,
+        validation_tests=[],
+    )
+    test_diffs = summary.file_changes
     assert test_diffs == diffs
```

Reply:
This leaves us freer to mess with the implementation details later down the line.

Reply:
Changed, thanks